exporter/datadogexporter/integrationtest/integration_test.go

```diff
@@ -642,10 +635,16 @@ func testIntegrationHostMetrics(t *testing.T, expectedMetrics map[string]struct{
 	// 2. Start in-process collector
 	factories := getIntegrationTestComponents(t)
 	app := getIntegrationTestCollector(t, "integration_test_host_metrics_config.yaml", factories)
+	var wg sync.WaitGroup
+	wg.Add(1)
 	go func() {
-		assert.NoError(t, app.Run(context.Background()))
+		_ = app.Run(context.Background()) // ignore shutdown error
```

Comment:
I think this needs some explanation in the code as to why this error is being ignored

Reply:
Done. This is https://github.com/open-telemetry/opentelemetry-collector/issues/12944